syncer/online-ddl-tools/online_ddl.go

```diff
@@ -488,11 +493,11 @@ func (r *RealOnlinePlugin) Finish(tctx *tcontext.Context, schema, table string)
 func (r *RealOnlinePlugin) TableType(table string) TableType {
 	// 5 is _ _gho/ghc/del or _ _old/new
 	if len(table) > 5 && strings.HasPrefix(table, "_") {
-		if strings.HasSuffix(table, "_gho") || strings.HasSuffix(table, "_new") {
+		if r.shadowReg.MatchString(table) {
```

Comment:
Remove the useless `if len(table) > 5 && strings.HasPrefix(table, "_")`, we should only depend on the pattern matching result here